pkg/graveler/ref/manager.go

```diff
@@ -397,21 +398,21 @@ func (m *Manager) addCommit(tx db.Tx, repositoryID graveler.RepositoryID, commit
 
 func (m *Manager) updateCommitGeneration(tx db.Tx, repositoryID graveler.RepositoryID, nodes map[graveler.CommitID]*CommitNode) error {
 	for len(nodes) != 0 {
-		command := "WITH updated(id, generation) AS (VALUES "
+		command := `WITH updated(id, generation) AS (VALUES `
 		var updatingRows int
-		for commitID, commitNode := range nodes {
+		for commitID, node := range nodes {
 			if updatingRows != 0 {
 				command += ","
 			}
-			command += fmt.Sprintf("('%s', %d)", commitID, commitNode.generation)
+			command += fmt.Sprintf(`(%s, %d)`, pq.QuoteLiteral(string(commitID)), node.generation)
 
 			delete(nodes, commitID)
 			updatingRows += 1
 			if updatingRows == BatchUpdateSQLSize {
 				break
 			}
 		}
-		command += fmt.Sprintf(") UPDATE graveler_commits SET generation = updated.generation FROM updated WHERE (graveler_commits.id=updated.id AND graveler_commits.repository_id='%s');", repositoryID)
+		command += fmt.Sprintf(`) UPDATE graveler_commits SET generation = updated.generation FROM updated WHERE (graveler_commits.id=updated.id AND graveler_commits.repository_id='%s');`, repositoryID)
```

Comment:
`"%s"` for `repository_id` is probably another SQL injection.  Luckily here we can just use standard SQL statement placeholders.  Look for "$1" just about anywhere for how to do that: it is the best and absolutely safest and simplest way to introduce strings into SQL statements.

Reply:
Thanks!

Looks very nice. One SQL injection vector remains and blocks approval.

I would also be happy if you got rid of the map in favour of a reference count. Right now I do not think it should block us; however if people do a commit every 10s then it may become significant. In any case we can do it on a separate PR.

Similarly, I'd really like for you to test the generations computation. That too does not need to happen in this PR.

@arielshaqed about the change of `parentsToVisit` to be counter- I'm using this map in mapCommitNodesToChildren so I can't really change it to be a counter
About the Generation calculation- we already have a test for it called `TestManager_FillGenerations` 😃 .

Forgot to answer those comments, sorry.